code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -181,22 +179,32 @@
 			"<span class='danger'>[L == user ? "You prod yourself" : "[user] has prodded you"] with [src]. Luckily it was off.</span>")
 		return
 
-	baton_stun(L, user)
-	user.do_attack_animation(L)
+	if(baton_stun(L, user))
+		user.do_attack_animation(L)
+
+/obj/item/melee/baton/proc/baton_stun(mob/living/L, mob/user, skip_cooldown = FALSE)
+	if(cooldown > world.time && !skip_cooldown)
```

Comment:
I'm not sure how this will feel ingame yet. But I can see it being a bit wacky. Not being able to hit somebody for 3.5 seconds on help mode. But you can on harm mode.
Maybe add a sound where the baton recharges? Maybe have the sprite blink or such

Reply:
telebatons have had a similar mechanic for a long time now. its not a major change and nobody has really complained since i moved it off of nextMove